backend/app/domain/services/utils/llm.py

```diff
@@ -510,6 +512,12 @@ async def generate_text(
         else:
             if is_llama == 1:
                 prompt = f"<s>[INST] {head_template} {prompt} {foot_template} [/INST]"
+            elif is_falcon == 1:
```

Comment:
Currently, we have a lot of 'if' conditions, which is fine for the moment. However, it might be a good idea to consider implementing a design pattern here.

Reply:
Makes sense. I don't think we'll get more than two other templates, if we do, I'll consider this possibility.